packages/stargate/src/aminomsgs.ts

```diff
@@ -331,7 +329,38 @@ export interface AminoMsgUndelegate extends AminoMsg {
 }
 
 export function isAminoMsgUndelegate(msg: AminoMsg): msg is AminoMsgUndelegate {
-  return (msg as AminoMsgUndelegate).type === "cosmos-sdk/MsgUndelegate";
+  return msg.type === "cosmos-sdk/MsgUndelegate";
 }
 
 // upgrade (no messages) - see https://github.com/cosmos/cosmos-sdk/blob/efa73c7/proto/cosmos/upgrade/upgrade.proto
+
+// ibc
+
+// https://github.com/cosmos/ibc-go/blob/07b6a97b67d17fd214a83764cbdb2c2c3daef445/modules/core/02-client/types/client.pb.go#L297-L312
+interface AminoHeight {
+  readonly revision_number: number;
+  readonly revision_height: number;
```

Comment:
For this type, the JSON serialization becomes a string: https://github.com/cosmos/cosmos-sdk/blob/v0.42.3/x/auth/legacy/legacytx/stdsign.go#L21-L22. Since there is no annotation for that, I assume uin64 -> string is the standard

Reply:
Hm, I was basing my choice on this: https://github.com/cosmos/cosmjs/blob/main/packages/stargate/src/aminomsgs.ts#L181

Reply:
But maybe that's a bug!

Reply:
It is very well possible that this is a bug.

We have in the LCD API:

```
export interface GovProposerResponse {
  readonly height: string;
  readonly result: {
    readonly proposal_id: string;
    readonly proposer: string;
  };
}
```

and in genesis

```
      "starting_proposal_id": "1",
```

Reply:
Any idea where the definitive resource for deciding this question might be?

Reply:
This is implemented here: https://github.com/tendermint/go-amino/blob/master/json-encode.go#L100-L113